### PR TITLE
External contracts

### LINKED ISSUE #41: Allow listening to arbitrary events emitted by addresses other than the proxy
Some apps need to get events from other addresses, for example the TokenManager app doesn't emit events on token transfers because the managed token already emits them.

This should also allow to pass a filter to take advantage of indexed event logs.

### PR DESCRIPTION
**Pull request by onbjerg**
This PR allows you to interact with external contracts (i.e. contracts that are assumed to not be an Aragon app), such as tokens.

It allows you to perform calls and filter for events, but provides no automatic caching or convenience functions.

**Usage-ish**

```js
const token = app.external(address, jsonInterface)

// A call
// Returns an Observable that emits a single value (the return value, a Web3 `Result` object)
token.symbol()

// Events
// Returns an Observable that emits Web3-structured events over time
token.events()
```

Note that you need to supply (at least a partial) JSON interface to interact with the contract.

**Example**

As a minimum viable example for getting the symbol of a token (conforming to Human Standard Token) and setting the app identifier to be that symbol, this would suffice:

```js
const Aragon = require('@aragon/client')

const app = new Aragon()
const token = app.external('0xdeadbeef', [
  {
    'constant': true,
    'inputs': [],
    'name': 'symbol',
    'outputs': [
      {
        'name': 'symbol',
        'type': 'string'
      }
    ],
    'payable': false,
    'type': 'function'
  }
])

token.symbol().subscribe(
  ({ symbol }) => app.identify(symbol)
)
```

**Improvements**

Some future improvements could include

- Transactions to external contracts, possibly with "universal transaction pathing"
- Convenience functions, such as an integration with `store` (see brief explanation in #75)

Closes #41 and partially adresses #43

### CODE REVIEW

**Comment by onbjerg** [comment]
Tested locally and working, need a go from either one of you @bpierre @sohkai

File: packages/aragon-client/src/index.js [hunk 1]
```diff
@@ -36,6 +36,45 @@ class AppProxy {
...
 
+  /**
+   * Create a handle to an external contract.
+   *
+   * @param  {string} address The address of the external contract
+   * @param  {Array<Object>} jsonInterface The JSON interface of the external contract
+   * @return {Object}
+   */
+  external (address, jsonInterface) {
+    let contract = {
```
Comment (sohkai) [inline packages/aragon-client/src/index.js]: Could be `const`?

File: packages/aragon-client/src/index.js [hunk 2]
```diff
@@ -36,6 +36,45 @@ class AppProxy {
...
+      events: (fromBlock = 0) => {
+        return this.rpc.sendAndObserveResponses(
+          'external_events',
+          [
+            address,
+            jsonInterface.filter(
+              (item) => item.type === 'event'
```
Comment (sohkai) [inline packages/aragon-client/src/index.js]: Cool! I had no idea you could put in partial ABIs and everything would just work.

File: packages/aragon-client/src/index.js [hunk 3]
```diff
@@ -36,6 +36,45 @@ class AppProxy {
...
+          ]
+        ).pluck('result')
+      }
+    }
+
+    // Bind calls
+    const callMethods = jsonInterface.filter(
+      (item) => item.type === 'function' && item.constant
+    )
+    for (let methodABI of callMethods) {
```
Comment (sohkai) [inline packages/aragon-client/src/index.js]: I do find it slightly odd this isn't a `forEach` or `reduce`

File: packages/aragon-client/src/index.js (hunk 1 above)
Comment (onbjerg) [reply packages/aragon-client/src/index.js]: yeah that's something I missed while iterating a bit

File: packages/aragon-client/src/index.js (hunk 3 above)
Comment (onbjerg) [reply packages/aragon-client/src/index.js]: could be a `forEach`, probably also a reduce but would feel a bit weird with reduce imo

**Pull request merged**